src/lib/lss.ts

```diff
@@ -239,3 +246,144 @@ export async function createSplitsXml(config: Config): Promise<string> {
+    .textContent?.trim();
+  let ordered: boolean | undefined;
+  if (orderedStr === "True") {
+    ordered = true;
+  } else if (orderedStr === "False") {
+    ordered = false;
+  }
+  const startTriggeringAutosplitStr =
+    autoSplitterSettings
+      .getElementsByTagName("AutosplitStartRuns")[0]
+      .textContent?.trim() || "";
+  let startTriggeringAutosplit: string | undefined;
+  if (startTriggeringAutosplitStr.length > 0) {
+    startTriggeringAutosplit = startTriggeringAutosplitStr;
+  }
+  // autosplitIds vs splitIds: autosplitIds do not contain "-" for subsplits, splitIds can
+  // parsedSplitIds is mutated as an object, but not as a variable
+  const parsedSplitIds: ParsedSplitId[] = [];
+  autoSplitterSettings
+    .getElementsByTagName("Splits")[0]
+    .childNodes.forEach((c) => {
+      if (c.nodeName === "Split") {
+        const autosplitId = c.textContent?.trim() || "";
+        parsedSplitIds.push({
+          autosplitId,
+          subsplit: false,
+          name: "",
+          iconId: "",
+        });
+      }
+    });
+  const endTriggeringAutosplit =
+    autoSplitterSettings
+      .getElementsByTagName("AutosplitEndRuns")[0]
+      .textContent?.trim() == "True";
+  // Segments Segment Name -> names, endingSplit name
+  const segments = xmlDoc
+    .getElementsByTagName("Segments")[0]
+    .getElementsByTagName("Segment");
+  // subsplitNames vs names: names do not contain "-" for subsplits, subsplitNames can
+  let endingSplitName = "";
+  for (let i = 0; i < segments.length; i++) {
+    const subsegmentName =
+      segments[i].getElementsByTagName("Name")[0].textContent?.trim() || "";
+    if (i < parsedSplitIds.length) {
+      if (subsegmentName.startsWith("-")) {
+        parsedSplitIds[i].subsplit = true;
+        parsedSplitIds[i].name = subsegmentName.substring(1);
+      } else {
+        parsedSplitIds[i].subsplit = false;
+        parsedSplitIds[i].name = subsegmentName;
+      }
+    } else if (i === parsedSplitIds.length) {
+      // endingSplit cannot be a subsplit
+      endingSplitName = subsegmentName;
+    }
+  }
+  let endingSplit: { name?: string } | undefined;
+  if (endingSplitName.length > 0) {
+    endingSplit = { name: endingSplitName };
+  }
+  // Deal with "-" subsplit markers
+  const splitIds: string[] = parsedSplitIds.map(({ autosplitId, subsplit }) => {
+    const namePrefix = subsplit ? "-" : "";
+    return `${namePrefix}${autosplitId}`;
+  });
+  // uniqueAutosplitIds is mutated as an object, but not as a variable
+  const uniqueAutosplitIds: string[] = [];
+  parsedSplitIds.forEach(({ autosplitId }) => {
+    if (!uniqueAutosplitIds.includes(autosplitId)) {
+      uniqueAutosplitIds.push(autosplitId);
+    }
+  });
+  const splitDefinitions = parseSplitsDefinitions();
+  // ns is mutated as an object, but not as a variable
+  const ns: Record<string, string | string[]> = {};
+  let anyNames = false;
+  for (let i = 0; i < uniqueAutosplitIds.length; i++) {
```

Comment:
loop body is not using index. also use a better name than `a`
```suggestion
  uniqueAutosplitIds.forEach((uniqueAutosplitId) => {
```